### PR TITLE
chore: change dependency management to poetry (#16)

### PR DESCRIPTION
**Pull request by vlavoie**
## Description

Poetry will be used from now on to match the other dialogue projects.
+isort & +pylama

## Related JIRA issues

Issue #1 

## Checklist

- [ ] PR title follows [Commit Convention](https://www.notion.so/godialogue/Commit-Convention-84fd9a4c149e48c998d760f1c9176df0) 
- [ ] PR title ends with a JIRA issue ID  
- [ ] All relevant PR sections are populated, irrelevant ones are removed 
- [ ] [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests) is opened for WIP changes 
- [ ] PR review requested from groups not individuals 
- [ ] PR link is posted to the corresponding Slack channel 

[DIA-123]: https://dialoguemd.atlassian.net/browse/DIA-123
[SOL-456]: https://dialoguemd.atlassian.net/browse/SOL-456

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -45,24 +37,21 @@ help:
 	@echo "        Start a Rasa shell using French models"
 
 init:
-	python3 -m venv ${VENV_LOCATION}
-	${ACTIVATE_VENV} pip install --upgrade pip
+	curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
```
Comment (vlavoie) [inline Makefile:40]: PS, je crois qu'il faudrait retourner à pip, c'est ça qu'ils font dans beautiful-mind et ça l'avait l'air de planter dans les pipelines (on en a pas ici par contre)

File: action-server/.vscode/settings.json [hunk 2]
```diff
@@ -16,9 +16,8 @@
   "python.linting.enabled": true,
   "python.linting.lintOnSave": true,
   "python.linting.pylintEnabled": false,
-  "python.linting.mypyEnabled": true,
+  "python.linting.mypyEnabled": false,
```
Comment (dmorand) [inline action-server/.vscode/settings.json:19]: Why are we disabling mypy?

File: action-server/.vscode/settings.json [hunk 3]
```diff
@@ -36,12 +35,11 @@
...
-    "tests"
-  ],
   "python.testing.unittestArgs": [
     "-v",
     "-s",
     "./tests"
-  ]
+  ],
+  "python.linting.pylamaEnabled": true,
+  "python.pythonPath": ".venv/bin/python3"
```
Comment (dmorand) [inline action-server/.vscode/settings.json:44]: Is this path correct?

File: action-server/Dockerfile [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
+ARG POETRY_VERSION="1.0.0"
+ENV \
+  PYTHONFAULTHANDLER=TRUE \
+  PYTHONUNBUFFERED=TRUE \
+  PIP_DISABLE_PIP_VERSION_CHECK=on \
+  PIP_NO_CACHE_DIR=off
+RUN \
+  pip install --upgrade pip \
+  && pip install "poetry==${POETRY_VERSION}" \
+  && poetry config virtualenvs.create false
```
Comment (dmorand) [inline action-server/Dockerfile:18]: Poetry should be installed before we copy the files to allow layer caching.

File: action-server/Dockerfile [hunk 5]
```diff
@@ -1,11 +1,24 @@
...
+RUN poetry install
 USER 1001
 
+FROM base as final
+
```
Comment (dmorand) [inline action-server/Dockerfile:23]: Aren't we missing copying the code? `COPY actions actions` should be performed in this build step.

File: action-server/Makefile [hunk 6]
```diff
@@ -4,12 +4,7 @@
 export PYTHONPATH = .
 
 VENV_LOCATION=.venv
-ACTIVATE_VENV=test -d ${VENV_LOCATION} \
-  && echo "Using .venv" \
-  && . ${VENV_LOCATION}/bin/activate;
-
-RASA_VERSION=1.9.4
-RASA_SDK_VERSION=1.9.0
+ACTIVATE_VENV=poetry run
```
Comment (dmorand) [inline action-server/Makefile:7]: At this point, I think we should inline this variable.

File: action-server/pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+[tool.poetry]
+name = "covidflow"
+version = "0.1.0"
+description = ""
+authors = [""]
+
+[tool.poetry.dependencies]
+pydantic = "^1.4"
+python = "^3.6"
+rasa-sdk = "^1.9.0"
```
Comment (dmorand) [inline action-server/pyproject.toml:10]: Is it problematic to "reinstall" the rasa-sdk in the rasa image?

File: core/pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,23 @@
+[tool.poetry]
+name = "covidflow"
+version = "0.1.0"
+description = ""
+authors = [""]
+
+[tool.poetry.dependencies]
+python = "^3.6"
+rasa-sdk = "^1.9.0"
```
Comment (dmorand) [inline core/pyproject.toml:9]: We should not be needing rasa-sdk here. We might eventually need rasa 1.9.4 tho.

File: action-server/Dockerfile (hunk 5 above)
Comment (vlavoie) [reply action-server/Dockerfile:23]: It was done in the base step, I will move it here.

File: action-server/pyproject.toml (hunk 7 above)
Comment (vlavoie) [reply action-server/pyproject.toml:10]: Doesn't seem to be a problem

File: action-server/.vscode/launch.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
+{
```
[the remark this answers is not in the document]
Comment (vlavoie) [reply action-server/.vscode/launch.json]: No

File: action-server/.vscode/settings.json (hunk 2 above)
Comment (vlavoie) [reply action-server/.vscode/settings.json]: I reenabled it

File: action-server/.vscode/settings.json (hunk 3 above)
Comment (vlavoie) [reply action-server/.vscode/settings.json]: Removed since vscode should be using the virtualenv and not a python path

**Pull request merged**